Finschia/finschia-sdk
# chore(deps): Bump Go from 1.20 to 1.22

## Description

closes: #XXXX

## Motivation and context



* Replace Go version of `go.mod`/Github action/Docker

## How has this been tested?




* **Test Environment** - Use 4 nodes for testing
  * Node 1, 2 : fnsad built by Go 1.20
  * Node 3, 4 : fnsad built by Go 1.22.4

* **Test Scenario**
  1. Set up 4 validators
  2. Connect four nodes each other (4P2)
  3. Send bank tx to Node1, Node3
  4. Verify that transactions are handled correctly and the statuses are queried well in Node2 and Node4
  5. Verify if there is any consensus breaking
 

## Screenshots (if appropriate):

## Checklist:



- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`
